### PR TITLE
Don't override target name when using self-signed certs

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
We were getting
```
Error getpublicclient config <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "Stream removed"
        debug_error_string = "{"created":"@1657750494.962014000","description":"Error received from peer ipv4:54.80.18.147:443","file":"src/core/lib/surface/call.cc","file_line":967,"grpc_message":"Stream removed","grpc_status":2}"
>
```
when using the admin client with a self-signed cert.

The reason is because the `cn` from the x509 subject had a * in it.  `*.elb.us-east-1.amazonaws.com` which was causing the ssl check to fail in nginx ingress.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2420

### CODE REVIEW

**Comment by brucearctor** [comment]
Great if that fixes this self-signed-cert/opta-setup issue….  But will this break others’ flows ( is that that override needed for others, ex: @RobertoRRW  who wrote the [override](https://github.com/flyteorg/flytekit/commit/065b8995c6246c55f8d1ba6fd2b74ee7c26314e9#diff-54d6ca794a724ff2e2b60e6dee730cb73865428d81f9e23c68e46e8fedf3ded6R128) in the first place )?

**Review by brucearctor — APPROVED** [review]
LGTM

File: flytekit/clients/raw.py [hunk 1]
```diff
@@ -119,13 +119,9 @@ def __init__(self, cfg: PlatformConfig, **kwargs):
                 server_address = tuple(endpoint_parts)
             else:
                 server_address = (cfg.endpoint, "443")
-
```
Comment (brucearctor) [inline flytekit/clients/raw.py:122]: what's style guide say about this?

Comment (wild-endeavor) [reply flytekit/clients/raw.py:122]: I just `make fmt` and what it doesn't change is not regulated I guess.

**Review by eapolinario — APPROVED** [review]

File: flytekit/clients/raw.py (hunk 1 above)
Comment (brucearctor) [reply flytekit/clients/raw.py:122]: Is there a published flyte styleguide though?  Yes, if `fmt` addresses that's great -- also good if I can write code that the formatter doesn't need to change :-)

**Review by eapolinario — APPROVED** [review]

File: flytekit/clients/raw.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/clients/raw.py:122]: No there is not unf.  Should we consider adding one?  I don't think I'd personally mind as long as it's a guide and not a strict set of rules.

Comment (brucearctor) [reply flytekit/clients/raw.py:122]: Suggesting that the guide would be the things that the formatter/linter would test for [ and reject if not adhered to ].  Anything more than that is getting too nitpicky.  I don't recall whether the project is using things like spotless, black/yapf, etc.  But, effectively a writeup of the things that these things enforce [ so people don't need to know/look-at the configurations of such tools ].

Comment (wild-endeavor) [reply flytekit/clients/raw.py:122]: oh yeah - `make fmt` just calls black.  and isort i think
